File: FT_PC/FT_PC/Util/FileHelper.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FT_PC
{
    class FileHelper
    {
        public static long receivedLength = 0;

        public static void setReceivedLength(long length)
        {
            receivedLength = receivedLength + length;
        }

        public static long getReceivedLength()
        {
            return receivedLength;
        }

        public static void resetreceivedLength()
        {
            receivedLength = 0;
        }

        /// <summary>
        /// 获得文件大小
        /// </summary>
        /// <param name="filePath">文件的路径</param>
        /// <returns></returns>
        public static long GetFileSize(String filePath)
        {
            long fileSize = 0L;
            if (filePath!="")
            {
                FileInfo fi = new FileInfo(filePath);
                fileSize = fi.Length; //返回文件字节大小
            }
           
            return fileSize;
        }

        /// <summary>
        /// 从文件路径中获得文件的名字
        /// </summary>
        /// <param name="filePath"></param>
        /// <returns></returns>
        public static String GetFileName(String filePath)
        {
            String fileName = null;

            int index = filePath.LastIndexOf("\\");
            fileName = filePath.Substring(index + 1, filePath.Length - 1 - index);

            return fileName;
        }
    }
}

File: FT/app/src/main/java/com/jcx/pc/util/FileHelper.java
package com.jcx.pc.util;

import android.os.Environment;

import java.io.File;
import java.io.IOException;

/**
 * Created by Cui on 16-6-8.
 */
public class FileHelper {

    public static long receivedLength = 0;

    public static void setReceivedLength(long length){
        receivedLength = receivedLength + length;
    }

    public static long getReceivedLength(){
        return receivedLength;
    }

    public static void resetreceivedLength(){
        receivedLength = 0;
    }

    /**
     * 获得文件的大小
     * @param filePath
     * @return
     */
    public static Long getFileSize(String filePath){
        Long fileLength = 0L;

        File file = new File(filePath);
        if (!file.exists()) {
            fileLength = file.length();
        }

        return fileLength;
    }

    /**
     * 新建接收到的文件，然后写入数据
     * @param fileName
     * @return
     */
    public static String newFile( String fileName){

        File receiveFile=null;

        String project_folder = Environment.getExternalStorageDirectory().getAbsolutePath()+"/"+"ft_pc";
        File parentFolder=new File(project_folder);
        File file=new File(project_folder+"/"+fileName);

        if (!parentFolder.exists()) {
            parentFolder.mkdir();
        }

        for (int i=1;file.exists();i++) {
            int index = fileName.lastIndexOf(".");
            String fileNameWithoutSuffix = fileName.substring(0,index);
            String suffix = fileName.substring(index+1,fileName.length());
            file=new File(project_folder+"/"+fileNameWithoutSuffix+ Integer.toString(i)+"."+suffix);
        }
        try {
            file.createNewFile();
        }  catch (IOException e) {
            e.printStackTrace();
        }
        receiveFile=file;

        return receiveFile.getAbsolutePath();
    }
}
